plugins/auth-backend/src/service/router.ts

```diff
@@ -59,7 +60,11 @@ export async function createRouter({
   });
   const catalogApi = new CatalogClient({ discoveryApi: discovery });
 
-  router.use(cookieParser());
+  const secret =
+    config.getOptionalString('auth.session.secret') ?? 'backstage secret';
```

Comment:
I'm thinking that we can make the session middleware be optional for now, so that it is only installed if a secret is provided through config. Please do add that `// TODO` about it being stateful as well

Reply:
Good call